widget/embedded/src/components/ConfirmWalletsModal/WalletList.tsx

```diff
@@ -73,55 +62,74 @@ export function WalletList(props: PropTypes) {
   const shouldShowMoreWallets = limit && numberOfSupportedWallets - limit > 0;
 
   const addExperimentalChain = async (wallet: Wallet) => {
-    setExperimentalChainWallet(null);
     setShowExperimentalChainModal(false);
+    setAddingExperimentalChainStatus('in-progress');
     await connect(wallet.walletType, wallet.chain);
+    setAddingExperimentalChainStatus('completed');
```

Comment:
I put a comment here to follow-up handling rejection case of adding experimental chains in another PR.